inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/instrumentation/hook/DispatchHookAdvices.java

```diff
@@ -0,0 +1,87 @@
+package rocks.inspectit.oce.core.instrumentation.hook;
+
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.dynamic.DynamicType;
+import net.bytebuddy.implementation.bytecode.assign.Assigner;
+import net.bytebuddy.matcher.ElementMatcher;
+import rocks.inspectit.oce.bootstrap.Instances;
+import rocks.inspectit.oce.bootstrap.context.IInspectitContext;
+import rocks.inspectit.oce.bootstrap.instrumentation.IMethodHook;
+
+import static net.bytebuddy.matcher.ElementMatchers.*;
+
+/**
+ * This class provides the hook implementations which are injected into specified classes in order to gather data.
+ *
+ * <p>
+ * IMPORTANT: Note that the implementation is inlined into the target application, thus, it have no access to the classes
+ * loaded by the inspectIT classloader! When needed, the classes provided in the bootstrap package have to be used!
+ */
+public class DispatchHookAdvices {
+
+
+    public static <T> DynamicType.Builder<T> adviceOn(DynamicType.Builder<T> builder, ElementMatcher<? super MethodDescription> methods) {
+        builder = builder.visit(Advice.to(NonStaticMethodAdvice.class).on(
+                not(isStatic()).and(not(isConstructor())).and(methods)));
+        builder = builder.visit(Advice.to(StaticMethodAdvice.class).on(isStatic().and(methods)));
+        builder = builder.visit(Advice.to(ConstructorAdvice.class).on(isConstructor().and(methods)));
+        return builder;
+    }
+
+    private static class NonStaticMethodAdvice {
+
+        @Advice.OnMethodEnter
+        public static void onEnter(@Advice.Origin Class<?> declaringClass, @Advice.Origin("#m#s") String signature,
```

Comment:
The descriptor is not what we need here:
https://docs.oracle.com/javase/specs/jvms/se7/html/jvms-4.html#jvms-4.3.3

And the return type is not included because it is actually not part of the signature, meaning that it doesn't help identifying a method.